### PR TITLE
Connecting the core array solver in strings

### PR DESCRIPTION
**Pull request by ajreynol**
This PR takes most of the remaining changes from the seqArray branch apart from the extension to model construction.

Notably it connects the core array solver to the array solver in strings.

### CODE REVIEW

File: src/expr/skolem_manager.h [hunk 1]
```diff
@@ -112,6 +112,8 @@ enum class SkolemFunId
    * i = 0, ..., n.
    */
   RE_UNFOLD_POS_COMPONENT,
+  /** Sequence model construction, character for base */
```
Comment (yoni206) [inline src/expr/skolem_manager.h:115]:
```suggestion
  /** Sequence model construction, element for base */
```

File: src/theory/strings/array_core_solver.h [hunk 2]
```diff
@@ -62,7 +62,7 @@ class ArrayCoreSolver : protected EnvObj
   const std::map<Node, Node>& getWriteModel(Node eqc);
 
   /**
-   * Get connected sequences
+   * Get connected sequences, see documentation above computeConnected.
```
Comment (yoni206) [inline src/theory/strings/array_core_solver.h:65]:
```suggestion
   * Get connected sequences, see documentation of computeConnected.
```

File: src/theory/strings/array_solver.h [hunk 3]
```diff
@@ -54,6 +55,33 @@ class ArraySolver : protected EnvObj
    * their application to concatenation terms.
    */
   void checkArrayConcat();
+  /**
+   * Perform reasoning about seq.nth and seq.update operations (lazily), which
+   * calls the core array solver for the set of nth/update terms over atomic
```
Comment (yoni206) [inline src/theory/strings/array_solver.h:60]:
```suggestion
   * calls the core sequences-array solver for the set of nth/update terms over atomic
```
Just so it doesn’t sound like we are calling the “real” AF_AX solver.

File: src/theory/strings/array_solver.h [hunk 4]
```diff
@@ -54,6 +55,33 @@ class ArraySolver : protected EnvObj
...
+   * equivalence classes.
+   */
+  void checkArray();
+  /**
+   * Same as above, but called eagerly, and for all nth/update terms, not just
```
Comment (yoni206) [inline src/theory/strings/array_solver.h:65]:
```suggestion
   * Same as `checkArray`, but called eagerly, and for all nth/update terms, not just
```

**Review by yoni206 — APPROVED** [review]
Thanks for this. Some minor suggestion inline, but otherwise LGTM.

File: src/theory/strings/array_solver.h [hunk 5]
```diff
@@ -54,6 +55,33 @@ class ArraySolver : protected EnvObj
...
+   * those over atomic equivalence classes.
+   */
+  void checkArrayEager();
+
+  /**
+   *
```
Comment (Ying1123) [inline src/theory/strings/array_solver.h:71]: This empty line could be deleted?

**Review by Ying1123 — APPROVED** [review]

**Pull request merged**